metrics/lib/common.bash

```diff
@@ -42,6 +42,47 @@ function save_results(){
 	bash $LIB_DIR/send_results.sh -n "$1" -a "$2" -r "$3" -u "$4"
 }
 
+# This function checks existence of commands.
+# They can be received standalone standalone or as an array, e.g.
+#
+# cmds=(“cmd1” “cmd2”)
+# check_cmds "${cmd[@]}"
+function check_cmds()
+{
+	req_tools="$@"
```

Comment:
This is incorrectly handling the arguments.

We should not be creating or altering global variables.

The correct thing to do IMNSHO is to use the short form of 'for',  i.e. 'for cmd ; do' and drop the req_tools as I showed in the previous comment.

If you are going to keep the req_tool, then you want to 
1. declare it to be local
1. you want to use an array assignment 
1. use the correct syntax to loop over the elements

As written, this code takes all the arguments, joins them together (usually with a space), and assigns this string to a scalar variable. It then treats this scalar as a one element array, takes all (one) element of this array (if there was more than one it would join them together with spaces), then splits again on spaces. It then splits apart again and loops over the words. All of this joining and splitting allows things to go wrong.

For this function, where you probably are not going to have spaces in the command names, you can get away with it, but in other functions you have the same pattern and do not know if you are going to have spaces in the data.

So this should say (I am leaving in 'function' to avoid the argument, although the links you provided are wrong, there is no alias issue in scripts)

    function check_cmds()
    {
        local cmd req_tools=( "$@" )
        for cmd in "${req_cmds[@]}"; do
            if ! 
    ....